### PR TITLE
Remove the make dependency

### PR DESCRIPTION
**Pull request by unfunco**
This commit removes the make dependency, since it seemed a roundabout
way of building the phar file, since the box.json finder ignore array
is also case-sensitive, I have added additional ignores from the
vendor directory, this reduces the size of the phar by about 20%.

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
Probably a good idea - Make was only really useful for building locally.

File: .travis.yml [hunk 1]
```diff
@@ -36,10 +36,10 @@ script:
 
 before_deploy:
   - curl -LSs https://box-project.github.io/box2/installer.php | php
-  - chmod +x box.phar
```
Comment (unfunco) [inline .travis.yml]: I think the installer does it automatically when piped to `php`, when I downloaded it and ran through the steps yesterday, `box.phar` was already executable. I could add it back in just incase box-project changes that in the future though?

Comment (unfunco) [inline .travis.yml]:
```
🍍  curl -LSs https://box-project.github.io/box2/installer.php | grep chmod
    @chmod($item->name, 0755);
```

Comment (ciaranmcnulty) [inline .travis.yml]: Ah! Good stuff

**Comment by ciaranmcnulty** [comment]
@unfunco can you rebase this against 2.5 please?

**Comment by ciaranmcnulty** [comment]
Good stuff, thanks!

**Pull request merged**